src/app/api/messages/services/message.service.ts

```diff
@@ -51,8 +51,9 @@ export class MessageService {
         await this.sendMessage(copilotClient, setting, message);
       }
     } catch (e: unknown) {
-      if (e instanceof ApiError) {
-        if (e.request?.errors && e.request.errors['404'] === '404') {
+      if (matchesCopilotApiError(e)) {
+        const castedErr = e as CopilotAPIError;
```

Comment:
```suggestion
        const castedErr = e;
```

Reply:
If you setup `matchesCopilotApiError` as a typeguard, you won't need to cast at all.